pkg/scheduler/apis/config/types.go

```diff
@@ -131,3 +140,63 @@ type KubeSchedulerLeaderElectionConfiguration struct {
 	// LockObjectName defines the lock object name
 	LockObjectName string
 }
+
+// Plugins include multiple extension points. When specified, the list of plugins for
+// a particular extension point are the only ones enabled. If an extension point is
+// omitted from the config, then the default set of plugins is used for that extension point.
+// If no plugin should be invoked at an extension point, an empty list must be provided.
+type Plugins struct {
+	// QueueSort is a list of plugins that should be invoked when sorting pods in the scheduling queue.
+	QueueSort []Plugin
+
+	// PreFilter is a list of plugins that should be invoked at "PreFilter" extension point of the scheduling framework.
+	PreFilter []Plugin
+
+	// Filter is a list of plugins that should be invoked when filtering out nodes that cannot run the Pod.
+	Filter []Plugin
+
+	// PostFilter is a list of plugins that should be invoked when a list of nodes that passed the filtering phase.
+	PostFilter []Plugin
+
+	// Score is a list of plugins that should be invoked when ranking nodes that have passed the filtering phase.
+	Score []Plugin
+
+	// NormalizeScore is a list of plugins that should be invoked once per plugin per scheduling cycle.
+	NormalizeScore []Plugin
+
+	// Reserve is a list of plugins that should be invoked when the last step in a scheduling cycle.
+	Reserve []Plugin
+
+	// Permit is a list of plugins that should be invoked when prevent or delay the binding of a Pod.
+	Permit []Plugin
+
+	// PreBind is a list of plugins that should be invoked before a pod is bound.
+	PreBind []Plugin
+
+	// Bind is a list of plugins that should be invoked at "Bind" extension point of the scheduling framework.
+	Bind []Plugin
+
+	// PostBind is a list of plugins that should be invoked after a pod is successfully bound.
+	PostBind []Plugin
+
+	// UnReserve is a list of plugins that should be invoked a pod was reserved and then rejected in a later phase.
+	UnReserve []Plugin
+}
+
+// Plugin specifies a plugin name and its weight when applicable. Weight is used only for Score plugins.
+type Plugin struct {
+	// Name defines the name of plugin
+	Name string
+	// Weight defines the weight of plugin, only used for Score plugins.
+	Weight int
+}
+
+// PluginConfig specifies arguments that should be passed to a plugin at the time of initialization.
+// A plugin that is invoked at multiple extension points is initialized once. Args can have arbitrary structure.
+// It is up to the plugin to process these Args.
+type PluginConfig struct {
+	// Name defines the name of plugin config
+	Name string
+	// Args defines the arguments of plugin config
```

Comment:
Args defines the arguments passed to the plugins at the time of initialization. Args can have arbitrary structure.